src/structure.js

```diff
@@ -487,21 +492,31 @@ function hasIn(cursor, path) {
   return cursor.getIn(path, NOT_SET) !== NOT_SET;
 }
 
-function pathString(path) {
-  var topLevel = 'global';
-  if (!path || !path.length) return topLevel;
-  return [topLevel].concat(path).join('|');
+function listListenerMatching (listeners, path) {
+  var pathArray = [],
+      matches = [];
+
+  path = path || [];
+
+  path.forEach(function(pathPart) {
+    pathArray = pathArray.concat(pathPart);
```

Comment:
What if pathPart here is a non-string? e.g. a complex structure (object, immutable structure, etc).